iot_verifier/src/poc.rs

```diff
@@ -406,6 +420,60 @@ fn verify_entropy(
     Ok(())
 }
 
+/// verify beacon construction
+fn verify_beacon_payload(
+    beacon_report: &IotBeaconReport,
+    region: ProtoRegion,
+    region_params: &[BlockchainRegionParamV1],
+    gain: i32,
+    entropy_start: DateTime<Utc>,
+    entropy_version: u32,
+) -> GenericVerifyResult {
+    // cast some proto based structs to beacon structs
+    let beacon_region: beacon::Region = region.into();
+    // TODO: support RegionParams::new in gateway rs
+    // avoid need to specify gain scale outside of region.rs
+    let beacon_region_params = beacon::RegionParams {
+        region: beacon_region,
+        params: region_params.to_owned(),
+        gain: Decimal::new(gain as i64, 1),
+    };
+    // generate a gateway rs beacon from the generated entropy and the beaconers region data
+    let generated_beacon = match generate_beacon(
+        &beacon_region_params,
+        entropy_start.timestamp_millis(),
+        entropy_version,
+        &beacon_report.local_entropy,
+        &beacon_report.remote_entropy,
+    ) {
+        Ok(res) => res,
+        Err(e) => {
+            tracing::warn!("failed to generate beacon, reason: {:?}", e);
+            return Err(InvalidReason::InvalidPacket);
+        }
+    };
```

Comment:
This can be written more concisely with a let else binding https://doc.rust-lang.org/nightly/rust-by-example/flow_control/let_else.html

Reply:
I was thinking `map_err` would make it easier to read
```
    let generated_beacon = generate_beacon(
        &beacon_region_params,
        entropy_start.timestamp_millis(),
        entropy_version,
        &beacon_report.local_entropy,
        &beacon_report.remote_entropy,
    )
    .map_err(|e| {
        tracing::warn!("failed to cast report to beacon, reason: {:?}", e);
        InvalidReason::InvalidPacket
    })?;
```

Reply:
used the map suggestion...